meta-mender-core/recipes-bsp/u-boot/patches/0001-Generic-boot-code-for-Mender.patch

```diff
@@ -40,55 +40,48 @@ index 0000000..3c1ecd6
 +
 +#include <config_mender_defines.h>
 +
-+#ifndef CONFIG_ENV_IS_IN_FAT
-+# error CONFIG_ENV_IS_IN_FAT is required for Mender to work
++#ifndef CONFIG_ENV_IS_IN_MMC
++# error CONFIG_ENV_IS_IN_MMC is required for Mender to work
 +#endif
 +
 +#ifndef CONFIG_BOOTCOUNT_LIMIT
 +# error CONFIG_BOOTCOUNT_LIMIT is required for Mender to work
 +#endif
 +
-+/* Currently Mender needs bootcount to reside in environment file. */
++/* Currently Mender needs bootcount to reside in environment. */
 +#ifndef CONFIG_BOOTCOUNT_ENV
 +# error CONFIG_BOOTCOUNT_ENV is required for Mender to work
 +#endif
 +
-+#ifdef FAT_ENV_INTERFACE
-+# error FAT_ENV_INTERFACE should not be defined explicitly (will be auto-configured).
++#ifdef CONFIG_SYS_MMC_ENV_DEV
++# error CONFIG_SYS_MMC_ENV_DEV should not be defined explicitly (will be auto-configured).
 +#endif
-+
-+/* Post-daisy variant. */
-+#ifdef FAT_ENV_DEVICE_AND_PART
-+# error FAT_ENV_DEVICE_AND_PART should not be defined explicitly (will be auto-configured).
-+#endif
-+/* Daisy-and-earlier variants. */
-+#ifdef FAT_ENV_DEVICE
-+# error FAT_ENV_DEVICE should not be defined explicitly (will be auto-configured).
++#ifdef CONFIG_SYS_MMC_ENV_PART
++# error CONFIG_SYS_MMC_ENV_PART should not be defined explicitly (will be auto-configured).
 +#endif
-+#ifdef FAT_ENV_PART
-+# error FAT_ENV_PART should not be defined explicitly (will be auto-configured).
++#ifdef CONFIG_ENV_OFFSET
++# error CONFIG_ENV_OFFSET should not be defined explicitly (will be auto-configured).
 +#endif
-+
-+#ifdef FAT_ENV_FILE
-+# error FAT_ENV_FILE should not be defined explicitly (will be auto-configured).
++#ifdef CONFIG_ENV_OFFSET_REDUND
++# error CONFIG_ENV_OFFSET_REDUND should not be defined explicitly (will be auto-configured).
 +#endif
 +
 +#if MENDER_BOOTENV_SIZE != CONFIG_ENV_SIZE
 +# error 'CONFIG_ENV_SIZE' define must be equal to bitbake variable 'BOOTENV_SIZE' set in U-Boot build recipe.
 +#endif
 +
-+#define FAT_ENV_INTERFACE       MENDER_UBOOT_STORAGE_INTERFACE
-+#define FAT_ENV_DEVICE_AND_PART __stringify(MENDER_UBOOT_STORAGE_DEVICE) ":" __stringify(MENDER_BOOT_PART_NUMBER)
-+#define FAT_ENV_DEVICE          MENDER_UBOOT_STORAGE_DEVICE
-+#define FAT_ENV_PART            MENDER_BOOT_PART_NUMBER
-+#define FAT_ENV_FILE            "uboot.env"
-+
-+#define CONFIG_FAT_WRITE
++#define CONFIG_SYS_MMC_ENV_DEV   MENDER_UBOOT_STORAGE_DEVICE
++/* PART in this context is not regular partition, but mmc partition, a
```

Comment:
Maybe change to `CONFIG_SYS_MMC_ENV_PART is in this ...`. It was hard to catch up what do you mean by part. Also, maybe just saying this is different thing you can add some comment what this is.